src/SurfaceSchemes/surface_setups.jl

```diff
@@ -0,0 +1,234 @@
+    thermo_params = CAP.thermodynamics_params(params)
+    ts = TD.PhaseNonEquil_pθq(thermo_params, p, θ, TD.PhasePartition(q_vap))
+    T = TD.air_temperature(thermo_params, ts)
+    shf = FT.(Dierckx.Spline1D(t_data, shf_data; k = 1))
+    lhf = FT.(Dierckx.Spline1D(t_data, lhf_data; k = 1))
+    # TODO: Replace Dierckx with a type-stable interpolation and remove the FT.
+    function surface_state(surface_coordinates, interior_z, t)
+        fluxes = HeatFluxes(; shf = shf(t), lhf = lhf(t))
+        parameterization = MoninObukhov(z0; fluxes, ustar)
+        return SurfaceState(; parameterization, T, p, q_vap)
+    end
+    return surface_state
+end
+
+struct DYCOMS_RF01 end
+function (::DYCOMS_RF01)(params)
+    FT = eltype(params)
+    T = FT(292.5)
+    p = FT(101780)
+    q_vap = FT(0.01384) # 0.01384 is taken from Pycles. TODO: Compute qstar(T).
+    z0 = FT(1e-4)
+    shf = FT(15)
+    lhf = FT(115)
+    parameterization = MoninObukhov(z0; fluxes = HeatFluxes(; shf, lhf))
+    return SurfaceState(; parameterization, T, p, q_vap)
+end
+
+struct DYCOMS_RF02 end
+function (::DYCOMS_RF02)(params)
+    FT = eltype(params)
+    T = FT(292.5)
+    p = FT(101780)
+    q_vap = FT(0.01384) # 0.01384 is taken from Pycles. TODO: Compute qstar(T).
+    z0 = FT(1e-4)
+    shf = FT(16)
+    lhf = FT(93)
+    ustar = FT(0.25)
+    parameterization = MoninObukhov(z0; fluxes = HeatFluxes(; shf, lhf), ustar)
+    return SurfaceState(; parameterization, T, p, q_vap)
+end
+
+struct Rico end
+function (::Rico)(params)
+    FT = eltype(params)
+    thermo_params = CAP.thermodynamics_params(params)
+    T_surface = FT(299.8)
+    p_surface = FT(101540)
+    z0 = FT(1.5e-4)
+    Cd = FT(0.001229)
+    Ch = FT(0.001094)
+    Cq = FT(0.001133) # TODO: Add support for Cq to SF.Coefficients.
+    # Saturated surface conditions for a given surface temperature and pressure
+    p_sat_surface =
+        TD.saturation_vapor_pressure(thermo_params, T_surface, TD.Liquid())
+    ϵ_v = TD.Parameters.R_d(thermo_params) / TD.Parameters.R_v(thermo_params)
+    q_surface = ϵ_v * p_sat_surface / (p_surface - p_sat_surface * (1 - ϵ_v))
+
+    function surface_state(surface_coordinates, interior_z, t)
+        # Adjust the coefficients from 20 m to the actual value of z.
+        adjustment = (log(20 / z0) / log(interior_z[1, 1, 1, 1, 1] / z0))^2
+        parameterization =
+            ExchangeCoefficients(Cd = Cd * adjustment, Ch = Ch * adjustment)
+        SurfaceState(;
+            parameterization,
+            T = T_surface,
+            p = p_surface,
+            q_vap = q_surface,
+        )
+    end
+    return surface_state
+end
+
+struct TRMM_LBA end
+function (::TRMM_LBA)(params)
+    FT = eltype(params)
+    T = FT(296.85)  # 0C + 23.7
+    p = FT(99130)
+    q_vap = FT(0.02245)
+    # Placeholder NaN, this value isn't actually used in SurfaceFluxes,
+    # but is required for the interface
+    z0 = FT(NaN)
```

Comment:
Is this value also not required for other cases like `DYCOMS_RF02`? If so, we should modify the interface to avoid passing around unused values.

Reply:
It looks like it's required for `DYCOMS_RF02` but not for `ARM_SGP`.